### PR TITLE
feature(cli): adding support for polling profile resources

### PR DESCRIPTION
**Pull request by xoscar**
This PR adds support for the polling profiles resource to be handled by the CLI

## Changes

- Adds the delete, get list and apply logic for polling profiles

## Fixes

- #2115 

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

https://www.loom.com/share/93b1b86b0e6c436295382ddbedd5dc10

### CODE REVIEW

File: cli/formatters/pollingprofile.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
+package formatters
+
```
Comment (xoscar) [inline cli/formatters/pollingprofile.go:2]: We need to refactor the way we handle formatters in the future

File: cli/cmd/config.go [hunk 2]
```diff
@@ -48,8 +48,11 @@ func setupCommand(options ...setupOption) func(cmd *cobra.Command, args []string
 
 		options := []actions.ResourceArgsOption{actions.WithClient(apiClient), actions.WithLogger(cliLogger), actions.WithConfig(cliConfig)}
 		configActions := actions.NewConfigActions(options...)
+		pollingActions := actions.NewPollingActions(options...)
 		resourceRegistry.Register(actions.SupportedResourceConfig, configActions)
 
+		resourceRegistry.Register(actions.SupportedResourcePollingProfile, pollingActions)
```
Comment (schoren) [inline cli/cmd/config.go:54]: Having constants for supported resources defined in the `actions` package makes this package a central dependency. You can just use regular strings defined as constants within each resource package (or file, whatever applies).

Keep in mind the usecase. no code is actually consuming these constants, only when the resource registers itself. There is no other direct reference to this constant, so it's only creating a hardcoded value that is not actually consumed

File: cli/cmd/delete_cmd.go [hunk 3]
```diff
@@ -1,21 +1,55 @@
...
+			return
+		}
+
+		cmd.Println(pterm.FgGreen.Sprintf(fmt.Sprintf("✔ Resource deleted successfully for resource type: %s", resourceType)))
 	},
 	PostRun: teardownCommand,
 }
 
 func init() {
+	deleteCmd.PersistentFlags().StringVarP(&deletedResourceID, "identifier", "i", "", "id of the resource to delete")
```
Comment (schoren) [inline cli/cmd/delete_cmd.go:53]: are you sure this needs to be a persistent flag? If i'm not mistaken, persistent flags are "inherited" by subcommands, while "not-persistent" flags are specific to the command only.

Comment (schoren) [reply cli/cmd/delete_cmd.go:53]: also, I think we defined the full flag to be `--id`, not `--identifier` (definition here: https://github.com/kubeshop/tracetest/issues/1937#issuecomment-1458493552). We probablly don't need a short version for the ID

File: cli/cmd/export_cmd.go [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+			return
+		}
+
+		cmd.Println(pterm.FgGreen.Sprintf(fmt.Sprintf("✔  Definition exported successfully for resource type: %s", resourceType)))
+	},
+	PostRun: teardownCommand,
+}
+
+func init() {
+	exportCmd.PersistentFlags().StringVarP(&exportResourceID, "identifier", "i", "", "id of the resource to export")
```
Comment (schoren) [inline cli/cmd/export_cmd.go:51]: I think we defined the full flag to be `--id`, not `--identifier` (definition here: https://github.com/kubeshop/tracetest/issues/1937#issuecomment-1458493552). We probablly don't need a short version for the ID

File: cli/cmd/list_cmd.go [hunk 5]
```diff
@@ -45,5 +59,9 @@ var listCmd = &cobra.Command{
 }
 
 func init() {
+	listCmd.PersistentFlags().Int32VarP(&listTake, "take", "t", 20, "Take number")
```
Comment (schoren) [inline cli/cmd/list_cmd.go:62]: Again, should these flags be persistent or not? Also, we can probablly skip the short version

File: cli/formatters/pollingprofile.go (hunk 1 above)
Comment (schoren) [reply cli/formatters/pollingprofile.go:2]: +1. This is a very hardcoded approach and won't scale to match the dynamic nature of the resource manager

File: server/model/yaml/encoding.go [hunk 6]
```diff
@@ -58,6 +59,13 @@ func Decode(contents []byte) (File, error) {
 			return File{}, fmt.Errorf("cannot decode config: %w", err)
 		}
 		f.Spec = config
+	case FileTypePollingProfile:
```
Comment (schoren) [inline server/model/yaml/encoding.go:62]: Is this consumed somewhere other than the CLI? If so, let's remove this and try to keep the new things decoupled from this old approach

**Review by schoren — CHANGES_REQUESTED** [review]
Nice work @xoscar!! The issue I see with this PR is that it's duplicating a lof of effor by making the client do all the type conversion, when the functionaliy is provided by the resourcemanager.

Some of the architectural goals of the resource manager are to make clients as thin as possible, and also reduce coupling between packages as much as possible.

File: server/resourcemanager/encoding.go [hunk 7]
```diff
@@ -65,5 +67,9 @@ func encoderFromRequest(r *http.Request) (encoder, error) {
 		}
 	}
 
+	if accept == "" && contentType == "" {
+		return defaultEncoder, nil
+	}
+
```
[the remark this answers is not in the document]
Comment (xoscar) [reply server/resourcemanager/encoding.go:73]: Let's talk about the whole content-type header thing, the autogenerated open API client for the CLI is causing some annoying things

File: cli/cmd/config.go (hunk 2 above)
Comment (xoscar) [reply cli/cmd/config.go:54]: Can I update this in the following PR? it should be about adding support for the demo resource

**Review by schoren — APPROVED** [review]
Looking great now. Thanks!

**Pull request merged**